src/Container/ApplicationFactory.php

```diff
@@ -85,13 +90,87 @@ public function __invoke(ContainerInterface $container)
     /**
      * Injects routes and the middleware pipeline into the application.
      *
-     * @param Application $app
-     * @param array $config
      * @return void
      */
-    private function injectRoutesAndPipeline(Application $app, array $config)
-    {
+    private function injectRoutesAndPipeline(
```

Comment:
I've added this method, and the other new methods in this factory, as this is the only place where _implicitly_ injecting the routing and dispatch middleware makes sense. Doing it from the `Application::pipeRoutingMiddleware()` (via the `ApplicationConfigInjectionTrait`) or the `ApplicationConfigInjectionDelegator` does not make sense, as these will typically execute _before_ the pipeline is fully built, leaving the pipeline in an incorrect state if any middleware are added via `config/pipeline.php` later.

In this particular factory, however, the methods only execute if the user has not opted into a programmatic pipeline; i.e., if they have specifically chosen to use configuration-driven pipelines, or not updated their pre-2.0 app to use programmatic pipelines. This is the only place implicit registration makes sense.